pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -174,9 +212,78 @@ func (s *GameServerSelector) Matches(gs *agonesv1.GameServer) bool {
 		}
 	}
 
+	if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		// Only check for matches if there are CounterSelectors or ListSelectors
+		if (s.Counters != nil || s.Lists != nil) && (len(s.Counters) != 0 || len(s.Lists) != 0) {
+			if !(s.matchCountsAndLists(gs)) {
+				return false
+			}
+		}
+	}
+
 	return true
 }
 
+func (s *GameServerSelector) matchCountsAndLists(gs *agonesv1.GameServer) bool {
+	match := false
+
+	// Match Counter Selectors
+	if s.Counters != nil && gs.Status.Counters != nil {
```

Comment:
Thinking about the case that `s.Counters` is not nil but `gs.Status.Counters` is nil, the match should be false. But in the current logic, match will be set to true by line 281 in this case if the required Lists got matched. Similar for the case where `s.Lists` is not nil but `gs.Status.Lists` is nil. Also better to have unit test coverage for these cases.

Reply:
Oh yep, good catch. I'll update.